R/prophet.R

```diff
@@ -0,0 +1,1189 @@
+  if (!ready) return()
+  
+  modelSummary         <- rstan::monitor(prophetModelResults$stan.fit, probs = c(0.025, 0.975))
+  modelSummaryRowNames <- rownames(modelSummary)
+  parNames             <- c("k", "m", "sigma_obs")
+  parSummary           <- modelSummary[modelSummaryRowNames %in% parNames, 1:ncol(modelSummary)]
+  parLabels            <- c("Growth rate (k)", "Offset (m)", "Residual variance (sigma)")
+  
+  prophetTable$addColumns(list(
+    par      = parLabels,
+    mean     = parSummary[["mean"]],
+    sd       = parSummary[["sd"]],
+    lowerCri = parSummary[["2.5%"]],
+    upperCri = parSummary[["97.5%"]],
+    rhat     = parSummary[["Rhat"]],
+    bulkEss  = parSummary[["Bulk_ESS"]],
+    tailEss  = parSummary[["Tail_ESS"]]
+  ))
+
+  return()
+}
+
+.prophetCreateChangePointTable <- function(jaspResults, options, ready) {
+  if (!is.null(jaspResults[["prophetMainContainer"]][["prophetChangePointTable"]]) || !options$changePointTable) return()
+
+  prophetModelResults <- jaspResults[["prophetResults"]][["prophetModelResults"]]$object
+  
+  title <- switch(options$estimation,
+                  map  = "Changepoint Estimates Table",
+                  mcmc = "Changepoint Posterior Summary Table")
+  prophetTable <- createJaspTable(title = title)
+  prophetTable$dependOn(c("changePointTable"))
+  prophetTable$position <- 3
+
+  prophetTable$addColumnInfo(name = "ds", title = gettext("Changepoint"), type = "string")
+  
+  if (options$estimation == "map") {
+    prophetTable$addColumnInfo(name = "delta", title = gettext("Change in growth rate (delta)"), type = "number")
+  } else {
+    parTitle <- gettext("Change in growth rate (delta)")
+    ciTitle <- gettext("95% Credible Interval")
+    prophetTable$addColumnInfo(name = "mean", title = gettext("Mean"), type = "number", overtitle = parTitle)
+    prophetTable$addColumnInfo(name = "sd", title = gettext("SD"), type = "number", overtitle = parTitle)
+    prophetTable$addColumnInfo(name = "lowerCri", title = gettext("Lower"), type = "number", overtitle = ciTitle)
+    prophetTable$addColumnInfo(name = "upperCri", title = gettext("Upper"), type = "number", overtitle = ciTitle)
+  }
+  
+  .prophetChangePointTableFill(prophetTable, prophetModelResults, options$estimation, ready)
+    
+  jaspResults[["prophetMainContainer"]][["prophetChangePointTable"]] <- prophetTable
+
+  return()
+}
+
+.prophetChangePointTableFill <-  function(prophetTable, prophetModelResults, estimation, ready) {
+  if (!ready) return()
+
+  delta  <- switch(estimation,
+                   map  = as.numeric(prophetModelResults$params$delta),
+                   mcmc = as.numeric(apply(prophetModelResults$params$delta, 2, mean)))
+  cps    <- as.character(prophetModelResults$changepoints)
+
+  if (estimation == "map") {
+    prophetTable$addColumns(list(
+      ds    = cps,
+      delta = delta
+    ))
+  } else {
+    deltaCri <- apply(prophetModelResults$params$delta, 2, quantile, probs = c(0.025, 0.975))
```

Comment:
Can we have these probabilities changeable as well?

Reply:
Yes, we can! I made that for the other tables and the parameter plots too!

Reply:
Are "params" and "delta" names of lists? If so, I'd recommend calling them directly/unambiguously via `prophetModelResults[["params"]][["delta"]]`

Reply:
Yes, I will do that from now on (and change the previous)!